src/main/java/com/indeed/jiraactions/TSVSpecBuilder.java

```diff
@@ -99,4 +127,44 @@ private static String getAllLinksValue(final Action action) {
 
         return String.join(" ", values);
     }
+
+    private static long getTotalStatusTime(final String statusType, final Action action) {
+        final List<StatusTime> st = action.getStatustimes();
+        long output = 0;
+        for(StatusTime statusTime : st) {
+            if (statusTime.getStatus().equals(statusType)) {
+                output = output + statusTime.getTimeinstatus();
+            }
+        }
+        return output;
+    }
+
+    private static long getTimeToFirst(final String statusType, final Action action) {
+        final List<StatusTime> st = action.getStatustimes();
+        long output = 0;
+        for(StatusTime statusTime : st) {
+            if (statusTime.getStatus().equals(statusType)) {
+                output = output + statusTime.getTimetofirst();
+            }
+        }
+        return output;
+    }
+
+    private static long getTimeToLast(final String statusType, final Action action) {
+        final List<StatusTime> st = action.getStatustimes();
+        long output = 0;
+        for(StatusTime statusTime : st) {
+            if (statusTime.getStatus().equals(statusType)) {
+                output = output + statusTime.getTimetolast();
+            }
+        }
+        return output;
+    }
+
+    private static String getAllStatuses(final Action action) {
+        final Iterable<String> values = action.getStatustimes().stream()
+                .map(StatusTime::getStatus)::iterator;
+
+        return String.join("|", values);
```

Comment:
You can do this in one line, something like
`action.getStatustimes().stream().map(StatusTime::getStatus).collect(Collectors.joining("|"));`